src/realm/dictionary.hpp

```diff
@@ -204,6 +205,108 @@ class Dictionary::Iterator : public ClusterTree::Iterator {
     Iterator(const Dictionary* dict, size_t pos);
 };
 
+// An interface used when the value type of the dictionary consists of
+// links to a single table. Implementation of the ObjList interface on
+// top of a Dictionary of objects. This is the dictionary equivilent of
+// LnkLst and LnkSet.
+class DictionaryLinkValues final : public ObjCollectionBase<CollectionBase> {
```

Comment:
Hmm, it's an interesting idea. The problem I have with putting all this link functionality into dictionary is that it clutters the API for all value types other than links. But I do like the reduction of boilerplate code, so there are tradeoffs. I think I would like to choose `DictionaryLinkValues` for the separation of concerns between other dictionary types and for consistency with the existing design of `LnkSet` and `LnkList`. Part of the problem is the query constructor taking a normal `Dictionary` vs taking a `DictionaryLinkValues`; I think having an explicit new type here can help readability to imply that only a certain type of dictionary can be used to create a query. If you strongly disagree, feel free to push back and I am open to being persuaded otherwise :)

Reply:
Ok. It is not a cardinal point for me. But you know - the fewer lines of code we write, the fewer we have to test.